### ISSUE TITLE
support multi-host / multi-tenant hazelcast cluster

### ISSUE THREAD

**Issue by poblin-orange** [hazelcast/hazelcast-cloudfoundry#1]
Hello,
glad to see support of hazelcast for cloudfoundry apps.

I understand that the broker you provide creates hazelcast single JVM services.
On our side (orange), we contributed a bosh release to cloudfoundry community, to provision an hazelcast cluster. https://github.com/cloudfoundry-community/hazelcast-bosh-uberjar

I think theres a possibility to merge your broker with our bosh release.
Do you think it makes sense ?
Are there multi-tenant capabilities in hazelcast that could be use to share a global cluster for n cloudfoundry service instances ?

**Comment by joshuamckenty** [comment]
We could look at using on-demand service provisioning to provide clusters for single-tenant use as well.

**Comment by poblin-orange** [comment]
@joshuamckenty Yes on demand bosh provisionning would be an option. I seen the side of cfsummit http://schd.ws/hosted_files/cfsummit2016/52/Alex%20Ley%20-%20On-Demand%20Service%20Provisioning%20with%20BOSH%202.0%20-%20DRAFT.pdf

Your suggest leveraging Bosh REST Api from the service broker?

**Comment by joshuamckenty** [comment]
Yes - but not directly, using the on-demand provisioning service instead. (It’s part of our partner SDK kit). Email me for details? jmckenty@pivotal.io

Or, you could talk directly to the BOSH API - but I recommend using BOSH 2.0 if you’re going to go that route.

**Comment by poblin-orange** [comment]
thks for your feedback. Agree Bosh 2 + some api calls seem to be the way to go, for a full oss solution.

**Comment by wildnez** [comment]
hello gents, thanks for the conversation and apologies for this massive delay in my response. @poblin-orange we are very interested in learning more about your Hazelcast bosh-release, can you please share your email id or a contact handle where I can reach you? Or if you wish, you could write to me at rahul@hazelcast.com.

Looking forward to hear from you soon.
